### PR TITLE
Enable Node 16 support

### PR DESCRIPTION
**Pull request by michael-booth**
Update to support build/testing & installing within node 16.

* Allow newer versions of eslint as a peer dep
* update GHA workflow to run on node 16
* Update dev dependencies to newer versions to allow use of eslint 7.x
* Various linting fixes

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "extends": "skyscanner",
+  "extends": "@skyscanner/skyscanner",
```
Comment (michael-booth) [inline .eslintrc.json:2]: eslint-skyscanner-config dep updated and is scoped now

File: package.json [hunk 2]
```diff
@@ -24,23 +24,21 @@
     ]
   },
   "devDependencies": {
+    "@skyscanner/eslint-config-skyscanner": "^11.1.0",
     "babel-eslint": "^10.0.1",
-    "ensure-node-env": "^1.0.8",
-    "eslint": "^5.15.3",
-    "eslint-config-airbnb": "^17.1.0",
-    "eslint-config-skyscanner": "^4.0.0",
```
Comment (michael-booth) [inline package.json:31]: eslint-config-airbnb is part of eslint-config-skyscanner so i don't see the need to install it separately.

eslint-config-skyscanner newer version is scoped hence it's removed here and re-added

File: package.json [hunk 3]
```diff
@@ -24,23 +24,21 @@
...
+    "eslint": "^7.32.0",
     "eslint-plugin-backpack": "^0.2.3",
-    "eslint-plugin-import": "^2.2.0",
-    "eslint-plugin-jsx-a11y": "^6.1.1",
-    "eslint-plugin-react": "^7.11.1",
+    "eslint-plugin-import": "^2.25.2",
+    "eslint-plugin-jsx-a11y": "^6.5.1",
+    "eslint-plugin-react": "^7.29.4",
```
Comment (michael-booth) [inline package.json:33]: Just dep updates to allow for eslint 7 as a peer dep

File: src/rules/messages.js [hunk 4]
```diff
@@ -1,3 +1,4 @@
 module.exports = {
-  deprecated: issue => `Deprecated ${issue}, use saddlebag-date package instead.`,
+  deprecated: (issue) =>
```
Comment (michael-booth) [inline src/rules/messages.js:2]: All the changes to .js files were generated using eslint --fix option.
Various lint errors were introduced after updating to newer versions of eslint-skyscanner-config

File: package.json [hunk 5]
```diff
@@ -24,23 +24,21 @@
     ]
   },
   "devDependencies": {
+    "@skyscanner/eslint-config-skyscanner": "^11.1.0",
     "babel-eslint": "^10.0.1",
```
Comment (jaysonwu991) [inline package.json:28]: Just a reminder here: `babel-eslint` has been deprecated, it recommends to use `@babel/eslint-parser`, but for now no need to switch.

File: package.json [hunk 6]
```diff
@@ -24,23 +24,21 @@
...
     "husky": "^1.0.1",
-    "jest": "^24.5.0",
+    "jest": "^27.5.1",
     "jsdoc": "^3.4.3",
     "jsdoc-to-markdown": "^4.0.1",
     "lint-staged": "^8.1.5"
   },
   "peerDependencies": {
-    "eslint": "^3.0.0 || ^4.0.0 || ^5.0.0"
+    "eslint": "^3.0.0 || ^4.0.0 || ^5.0.0 || ^6.0.0 || ^7.0.0"
```
Comment (jaysonwu991) [inline package.json:41]:
```suggestion
    "eslint": "^5.0.0 || ^6.0.0 || ^7.0.0"
```
For version 5, it has been 3 years ago when it was first released, for security concern, I think version 3 and version 4 should be abandoned.

**Review by jaysonwu991 — APPROVED** [review]
A good job, and I recommend to add `README.md` for releasing, the change is in `package.json` file:
```json
"files": [
    "src"
  ]
```
-->
```json
"files": [
    "src",
    "README.md"
  ]
```
For it can give a guidance for engineers directly in `node_modules` folder and there is no need to jump to this repository for that.

File: package.json (hunk 6 above)
Comment (michael-booth) [reply package.json:41]: That sounds reasonable but we would need to consider it a breaking change if we do that.

I suggest we drop 3.x & 4.x as part of the next major version release which will be happening when Sophia makes her changes to recommend using date-fns

**Pull request merged**